native-token/src/lib.rs

```diff
@@ -40,3 +50,74 @@ impl Debug for Sol {
         self.write_in_sol(f)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_lamports_to_sol() {
+        assert_eq!(0.0, lamports_to_sol(0));
+        assert_eq!(0.000000001, lamports_to_sol(1));
+        assert_eq!(0.00000001, lamports_to_sol(10));
+        assert_eq!(0.0000001, lamports_to_sol(100));
+        assert_eq!(0.000001, lamports_to_sol(1000));
+        assert_eq!(0.00001, lamports_to_sol(10000));
+        assert_eq!(0.0001, lamports_to_sol(100000));
+        assert_eq!(0.001, lamports_to_sol(1000000));
+        assert_eq!(0.01, lamports_to_sol(10000000));
+        assert_eq!(0.1, lamports_to_sol(100000000));
+        assert_eq!(1., lamports_to_sol(1000000000));
+        assert_eq!(4.1, lamports_to_sol(4_100_000_000));
+        assert_eq!(8.2, lamports_to_sol(8_200_000_000));
+        assert_eq!(8.50228288, lamports_to_sol(8_502_282_880));
+        assert_eq!(18446744073.70955276489257812500, lamports_to_sol(u64::MAX));
```

Comment:
consider
#[allow(clippy::excessive_precision)]
for the lines like this or checks will fail
@svenski123

Reply:
Putting that attribute on the floating point literals causes `error[E0658]: attributes on expressions are experimental` errors so have put them on the test functions instead.